File: src/app/model/plantaUsuario.ts
import { Planta } from "./planta";
import { Usuario } from "./usuario";

export class PlantaUsuario {
    id: number;
    idPlanta: number;
    idUsuario: number;
    descricao: string;
    nomeUsuario: string;
    avaliacao: number;
    arquivoFotoUsuario: string;


    constructor() {
        this.id = 0;
        this.idPlanta = 0;
        this.idUsuario = 0;
        this.descricao = "";
        this.nomeUsuario = "";
        this.avaliacao = 0;
        this.arquivoFotoUsuario="";
    

    }
}

File: API/src/main/java/br/cefet/trabalho/model/PlantaUsuario.java
package br.cefet.trabalho.model;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class PlantaUsuario {
	private int id;
	 
	private int idPlanta;  
    private int idUsuario;  

    @NotBlank(message = "")
    @Size(min = 2, max = 255)
    private String descricao;

    private int avaliacao;
    
    private String arquivoFotoUsuario; 
}